### PR TITLE
[NativeAOT] Unify common cases of Array.Copy with CoreCLR

### PR DESCRIPTION
**Pull request by VSadov**
NativeAOT has the same common cases of Array.Copy, modulo couple `EETypePtr` APIs not present on `MethodTable`

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/EETypePtr.cs [hunk 1]
```diff
@@ -361,7 +361,7 @@ internal bool HasPointers
         {
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/EETypePtr.cs:361]: Could you please rename EETypePtr.HasPointers too for consistency?

File: src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/MethodTable.Runtime.cs [hunk 2]
```diff
@@ -98,6 +98,14 @@ internal static bool BothSimpleCasting(MethodTable* pThis, MethodTable* pOther)
             return ((pThis->_uFlags | pOther->_uFlags) & (uint)EETypeFlags.ComplexCastingMask) == 0;
         }
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/MethodTable.Runtime.cs:99]: You can delete FastEquals on EETypePtr. Array.Copy was the only method that used it.

File: src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/MethodTable.Runtime.cs [hunk 3]
```diff
@@ -98,6 +98,14 @@ internal static bool BothSimpleCasting(MethodTable* pThis, MethodTable* pOther)
...
 
+        internal static bool FastEquals(MethodTable* mt1, MethodTable* mt2)
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/MethodTable.Runtime.cs:101]: I would give this a better name, like `static bool AreSameType(MethodTable* mt1, MethodTable* mt2)`.

Comment (VSadov) [reply src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/MethodTable.Runtime.cs:102]: `AreSameType` sounds better. No need to explain why no equivalence checks on CoreClr side.

**Review by jkotas — APPROVED** [review]

**Comment by VSadov** [comment]
Thanks!!

**Pull request merged**